src/thumb/ucb.rs

```diff
@@ -0,0 +1,37 @@
+use crate::get_bits;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+struct UnconditionalBranch { 
+    offset11: u16,
+}
+
+impl From<u16> for UnconditionalBranch { 
+    fn from(raw: u16) -> Self { 
+        debug_assert!(get_bits!(raw, 15..11) == 0b11100);
+
+        Self { 
+            offset11: get_bits!(raw, 10..0)
+        }
+    }
+}
+
+
+#[cfg(test)]
+mod test { 
+    use crate::thumb::ucb::UnconditionalBranch;
+    use proptest::prelude::*;
+
+    proptest! { 
+        #[test]
+        fn props(offset11 in 0_u16..0b11111111111) {
```

Comment:
I recommend 0..(1 << 11) for two reasons - first, this range is exclusive - so it doesn't test the last value. Second, the number of intended bits is easier to read - it's clearly testing 11 bits, which matches the number in the spec. Currently, I have to individually count the number of 1s, which is error prone and slightly tiring.